lms/djangoapps/philu_overrides/tests/test_context_processor.py

```diff
@@ -0,0 +1,590 @@
+"""
+Tests for context processors in philu_overrides
+"""
+
+import mock
+import pytest
+from django.conf import settings
+from django.test import RequestFactory
+from django.urls import reverse
+
+from lms.djangoapps.philu_overrides.constants import (
+    ACTIVATION_ALERT_TYPE,
+    ACTIVATION_ERROR_MSG_FORMAT,
+    ORG_DETAILS_UPDATE_ALERT,
+    ORG_OEF_UPDATE_ALERT
+)
+from lms.djangoapps.philu_overrides.context_processor import (
+    add_nodebb_endpoint,
+    get_cdn_link,
+    get_global_alert_messages
+)
+from student.tests.factories import AnonymousUserFactory, UserFactory
+
+
+@pytest.fixture()
+def request_object(db):  # pylint: disable=unused-argument
+    request = RequestFactory().get('/dummy_url')
+    request.user = UserFactory.create()
+    return request
+
+# Tests for get_global_alert_messages context processor
+
+# ACTIVATION_ERROR_MSG Tests
+
+
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_detail_platform_overlay_available', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_oef_prompt_available', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.get_org_oef_update_prompt', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_detail_prompt_available', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.get_org_metric_update_prompt', mock.Mock())
```

Comment:
Since all of these are repeating in almost all tests can we avoid rewriting it again? You can create a fixture for common mocks and use it on relavent test. There is a library [pytest-mock](https://pypi.org/project/pytest-mock/), we can use it for tests. It will significantly reduce the following code lines and complexity. Please discuss library with @ahmed-faraz46 

```
@pytest.fixture()
def mock_get_global_alert_messages_dependencies():
    with mock.patch(
        'lms.djangoapps.philu_overrides.context_processor.is_org_detail_platform_overlay_available') as a, \
        mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_oef_prompt_available') as b, \
        mock.patch('lms.djangoapps.philu_overrides.context_processor.get_org_oef_update_prompt') as c, \
        mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_detail_prompt_available') as d, \
        mock.patch('lms.djangoapps.philu_overrides.context_processor.get_org_metric_update_prompt') as e:
        yield a, b, c, d, e


def test_get_org_oef_update_alert_message_success(request_object, mock_get_global_alert_messages_dependencies):

    _, mock_is_org_oef_prompt_available, mock_get_org_oef_update_prompt, _, _ = mock_get_global_alert_messages_dependencies

    mock_get_org_oef_update_prompt.return_value = True
    mock_is_org_oef_prompt_available.return_value = True

```